File: internal/domain/entity/dungeon.go
package entity

import "time"

type Dungeon struct {
	ID             string
	Title          string
	AdminUserID    int64
	TelegramChatID *int64 // Optional link to Telegram group
	CreatedAt      time.Time
}

type DungeonMember struct {
	DungeonID string
	UserID    int64
	JoinedAt  time.Time
}

File: frontend/src/types/dungeon.ts
export interface Dungeon {
  id: string;
  title: string;
  admin_user_id: number;
  telegram_chat_id?: number;
  created_at: string;
}

export interface CreateDungeonRequest {
  title: string;
  telegram_chat_id?: number;
}

export interface DungeonSummary {
  balance: string;
  todays_quests: any[];
  recent_purchases: any[];
}